simple-wallet/src/lib.rs

```diff
@@ -171,55 +177,32 @@ where
     }
 }
 
-impl<B: Deref, W: Deref> Signer for SimpleWallet<B, W>
+/// Computes the secret key from the temporary id and the seed.
+pub(crate) fn compute_secret_key(seed: [u8; 32], keys_id: [u8; 32]) -> SecretKey {
+    let mut hmac = HmacEngine::<Sha256>::new(&seed);
+    hmac.input(&keys_id);
+    let secret_bytes = Hmac::from_engine(hmac).into_inner();
+    SecretKey::from_slice(&secret_bytes).expect("Secret key is valid")
+}
+
+impl<B: Deref, W: Deref> ContractSignerProvider for SimpleWallet<B, W>
 where
     B::Target: WalletBlockchainProvider,
     W::Target: WalletStorage,
 {
-    fn sign_psbt_input(
-        &self,
-        psbt: &mut PartiallySignedTransaction,
-        input_index: usize,
-    ) -> std::result::Result<(), Error> {
-        let tx_out = if let Some(input) = psbt.inputs.get(input_index) {
-            if let Some(wit_utxo) = &input.witness_utxo {
-                Ok(wit_utxo.clone())
-            } else if let Some(in_tx) = &input.non_witness_utxo {
-                Ok(
-                    in_tx.output[psbt.unsigned_tx.input[input_index].previous_output.vout as usize]
-                        .clone(),
-                )
-            } else {
-                Err(Error::InvalidParameters(
-                    "No TxOut for PSBT inout".to_string(),
-                ))
-            }
-        } else {
-            Err(Error::InvalidParameters(
-                "No TxOut for PSBT inout".to_string(),
-            ))
-        }?;
-        let address = Address::from_script(&tx_out.script_pubkey, self.network)
-            .expect("a valid scriptpubkey");
-        let seckey = self
-            .storage
-            .get_priv_key_for_address(&address)?
-            .expect("to have the requested private key");
+    type Signer = SimpleSigner;
 
-        let mut tx = psbt.unsigned_tx.clone();
-        dlc::util::sign_p2wpkh_input(
-            &self.secp_ctx,
-            &seckey,
-            &mut tx,
-            input_index,
-            bitcoin::EcdsaSighashType::All,
-            tx_out.value,
-        )?;
+    fn derive_signer_key_id(&self, _is_offer_party: bool, temp_id: [u8; 32]) -> [u8; 32] {
+        // fixme: probably not the best
+        let secret_key = compute_secret_key(self.seed, temp_id);
+        secret_key.secret_bytes()
+    }
 
-        let tx_input = tx.input[input_index].clone();
-        psbt.inputs[input_index].final_script_sig = Some(tx_input.script_sig);
-        psbt.inputs[input_index].final_script_witness = Some(tx_input.witness);
-        Ok(())
+    fn derive_contract_signer(&self, keys_id: KeysId) -> Result<Self::Signer> {
+        let secret_key = compute_secret_key(self.seed, keys_id);
```

Comment:
I feel the best would be to just generate a new secret key and associate it with the `keys_id` in storage (might be a bit annoying to change the storage interface but feels cleaner)